src/v/coproc/tests/pacemaker_tests.cc

```diff
@@ -232,17 +232,31 @@ FIXTURE_TEST(test_copro_delete_topic, coproc_test_fixture) {
     info("All produced onto: {}", tbd);
     f.get();
 
-    /// Wait until all data is recieved on output topic, which should have been
-    /// recreated and fully processed from the earliest offset
-    for (auto i = 0; i < 4; ++i) {
-        auto results = consume(
-                         model::ntp(
-                           tbd.ns,
-                           to_materialized_topic(
-                             tbd.tp, identity_coprocessor::identity_topic),
-                           model::partition_id(i)),
-                         total_records)
-                         .get();
-        BOOST_CHECK_EQUAL(num_records(results), total_records);
+    /// Theres no synchonrization between deletion of materialized topic and
+    /// completion of processing by the coproc fiber. Following assert depends
+    /// on the condition
+    auto was_deleted = root_fixture()->app.cp_partition_manager.invoke_on(
+      *shard,
+      [target](coproc::partition_manager& pm) { return (bool)pm.get(target); });
+    if (was_deleted.get()) {
+        auto in_storage = root_fixture()->app.storage.invoke_on(
+          *shard,
+          [target](storage::api& api) { return api.log_mgr().get(target); });
+        BOOST_CHECK(!in_storage.get());
+    } else {
+        info("Materialized topic {} was re-created", target);
+        /// Wait until all data is recieved on output topic, which should have
+        /// been recreated and fully processed from the earliest offset
+        for (auto i = 0; i < 4; ++i) {
+            auto results = consume(
+                             model::ntp(
+                               tbd.ns,
+                               to_materialized_topic(
+                                 tbd.tp, identity_coprocessor::identity_topic),
+                               model::partition_id(i)),
+                             total_records)
+                             .get();
```

Comment:
Can  deletion be between consume command? Looks like in this case we will have the same problem

Reply:
Your right in the logic hasn't changed, but the assertion has. Check out the conditional that has been introduced.

Reply:
I think you might be right, i am going to shutdown the fiber then before the conditional